main.go

```diff
@@ -91,6 +105,18 @@ func main() {
 	// Parse command-line arguments.
 	args := argsParse(nil)
 
+	// Read the config file
+	err := configParse(args.configFile)
+	if err != nil {
+		stderrLogger.Printf("Failed to read config file: %v", err)
+	}
+
+	// Parse config overrides from command line
+	err = yaml.Unmarshal([]byte(args.options), &config)
```

Comment:
so you're unmarshalling identically formatted data using the same function twice: here and [here](https://github.com/kubernetes-incubator/node-feature-discovery/pull/146/files#diff-7ddfb3e035b42cd70649cc33393fe32cR224) - maybe there's a way to refactor this to make it more "dry"?

Reply:
I'm not sure how to do what you're after. We want to separate the cases where config file parsing and --options parsing fails. That is, we want to parse twice, and have two distinct errors

Reply:
It's a minor issue, but maybe something like that (pseudo-code):
```
configFileContent ,err := readConfigFile(path)
//if err
err = parseConfig(configFileContent)
//if err
err = parseConfig([]byte(args.options)
//if err
```
in order to have a single place in the code where you parse the config []byte. And you still handle errors from both sources. Imagine something in the future changes and you have to recreate the config parsing logic from YaML to INI (bizarre turn of events :wink:) - this way you would need to rework only single function instead of two.